src/plugins/home/server/services/sample_data/data_sets/otel/index.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { i18n } from '@osd/i18n';
+import path from 'path';
+import { AppLinkSchema, SampleDatasetSchema } from '../../lib/sample_dataset_registry_types';
+import {
+  appendDataSourceId,
+  getSavedObjectsWithDataSource,
+  overwriteSavedObjectsWithWorkspaceId,
+} from '../util';
+// import { getSavedObjects } from './saved_objects';
+import { servicesFieldMappings } from './services_field_mappings';
+import { tracesFieldMappings } from './traces_field_mappings';
+
+const otelDataName = i18n.translate('home.sampleData.otelSpecTitle', {
+  defaultMessage: 'Sample Open-Telemetry logs, Traces and Metrics',
+});
+const otelDataDescription = i18n.translate('home.sampleData.otelSpecDescription', {
+  defaultMessage:
+    'Sample data including correlated observability signals for an e-commerce application in Open-Telemetry standard.',
+});
+const initialAppLinks = [] as AppLinkSchema[];
+
+const DEFAULT_INDEX = 'i254a9691-8a1b-11ef-adb4-9f1097200c20';
+const DASHBOARD_ID = '';
+
+export const otelSpecProvider = function (): SampleDatasetSchema {
+  return {
+    id: 'otel',
+    name: otelDataName,
+    description: otelDataDescription,
+    previewImagePath: '/plugins/home/assets/sample_data_resources/otel/otel_traces.png',
+    darkPreviewImagePath: '/plugins/home/assets/sample_data_resources/otel/otel_traces_dark.png',
+    hasNewThemeImages: true,
+    overviewDashboard: DASHBOARD_ID,
```

Comment:
Shall we update `SampleDatsetSchema` to set field to optional? Since this `overviewDashboard` value will be used in [this line](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/home/public/application/components/sample_data_view_data_button.js#L71), do we need to hide this button or show another link?

Reply:
Didn't want to keep this optional, rather just kept it empty for now. Similar to how applinks are kept empty in other sample data objects. Have hidden the button for dashbords if empty and showing app links now. 
<img width="1520" alt="Screenshot 2024-10-18 at 1 02 20 PM" src="https://github.com/user-attachments/assets/3eb27407-3fcf-4849-8261-a3023d014fd8">